interpreter/cling/include/cling/Utils/ASTImportSource.h

```diff
@@ -0,0 +1,88 @@
+#ifndef CLING_ASTIMPORTSOURCE_H
+#define CLING_ASTIMPORTSOURCE_H
+
+#include "cling/Interpreter/Interpreter.h"
+
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Sema/Sema.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/AST/ASTImporter.h"
+
+#include <string>
+#include <map>
+
+namespace clang {
+  class ASTContext;
+  class Expr;
+  class Decl;
+  class DeclContext;
+  class DeclarationName;
+  class GlobalDecl;
+  class FunctionDecl;
+  class IntegerLiteral;
+  class NamedDecl;
+  class NamespaceDecl;
+  class NestedNameSpecifier;
+  class QualType;
+  class Sema;
+  class TagDecl;
+  class TemplateDecl;
+  class Type;
+  class TypedefNameDecl;
+}
+
+namespace cling {
+  namespace utils {
+
+    class ASTImportSource : public clang::ExternalASTSource {
+
+      private:
+        cling::Interpreter *m_first_Interp;
+        cling::Interpreter *m_second_Interp;
+        const clang::TranslationUnitDecl *m_translationUnitI1;
+        const clang::TranslationUnitDecl *m_translationUnitI2;
+
+        clang::Sema *m_Sema;
+        /* A map for the DeclContext-to-DeclContext correspondence
```

Comment:
Can you make that comment Doxygen style? Can you state that key is second interpreter and value is first interpreter (if that's what it is)? That's really essential information for this map - the fact that it's a map of decl context to decl context is clearly visible from the type - but you ought to explain what it's meaning is. You should probably even state why you have it. All of that should still only be one our two sentences :-) This is the most important member of your class!